git-diff/action.yml

```diff
@@ -14,43 +18,141 @@ outputs:
+      uses: actions/checkout@v4
+      with:
+        repository: trakx/github-actions
+        path: ./github-actions-git-diff
+        ref: ${{ inputs.actionsRepoRef }}
 
-    - name: "Detect Changes"
-      id: files
-      uses: jitterbit/get-changed-files@v1
+    - name: Install GH CLI
+      uses: ./github-actions-git-diff/install-github-cli
+
+    # When the event is a workflow_run, we need to download the artifact
+    # This artifact is being created by test-dotnet action, since this action usually triggers other workflows
+    - name: "Load event commits from artifact (when event is workflow_run)"
+      id: getArtifactCommits
+      if: ${{ github.event_name == 'workflow_run' }}
+      uses: ./github-actions-git-diff/load-event-commits-artifact
       with:
-        format: 'json'
+        workflowRunId: ${{ github.event.workflow_run.id }}
+
+    - name: "Detect changes from event commits"
+      id: files
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+        GH_TOKEN: ${{ github.token }}
+      shell: bash
+      run: |
+        # first try to get the commits from the current workflow event
+        commitShas='${{ join(github.event.commits.*.id, ' ') }}'
+
+        # if has no commits from the current workflow event, should use the commits from the artifact, which should be loaded from the "caller" workflow
+        if [[ -z $commitShas ]]; then
+          echo "[DEBUG] No commits found in this workflow event, using commits from the artifact"
+          commitShas="${{ steps.getArtifactCommits.outputs.artifactCommits }}"
+        fi
+
+        echo "[DEBUG] Event commits found: $commitShas"
+
+        # ensure commitShas has at least one space
+        if [[ $commitShas != *" "* ]]; then
+          commitShas="$commitShas "
+        fi
+
+        # split commitShas by space
+        IFS=' ' read -ra commitShas <<< "$commitShas"
+
+        FILES=""
+
+        for sha in $commitShas; do
+          echo "[DEBUG] Loading files from commit $sha"
+
+          commits=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "/repos/${{ github.repository }}/commits/${sha}")
+          filenames=$(jq --join-output '.files[] | select(.status == "modified" or .status == "added") | "\(.filename) "' <<<"${commits}")
+          for filename in $filenames; do
+            # concat fileName to FILES split by comma if not already there
+            if [[ $FILES != *"$filename"* ]]; then
+              FILES="$filename,$FILES"
+            fi
+          done
+        done
+        FILES=${FILES::-1}
+        echo "changedFiles=$FILES" >> $GITHUB_OUTPUT
+        echo "commitShas=$commitShas" >> $GITHUB_OUTPUT
 
     - name: "Evaluate Changes that are under the inputs.paths (split by comma)"
       id: evaluateStep
       shell: bash
       run: |
-        IFS=',' read -ra paths <<< "${{ inputs.paths }}"
+        inputPaths="${{ inputs.paths }}"
```

Comment:
Some fixes were also made in this step to ensure it iterates all files and paths.